ImageFilterEditor/ImageFilterEditor/ViewController.swift

```diff
@@ -0,0 +1,368 @@
+            sliderB.maximumValue = Float((scaledImage?.size.height)!)
+            sliderB.value = Float((scaledImage?.size.height)!) / 2
+
+            sliderC.minimumValue = 0
+            sliderC.maximumValue = 100
+            sliderC.value = 20
+
+
+        case 1:
+            for i in 0..<allLabels.count {
+                allLabels[i]?.isHidden = false
+                allSliders[i]?.isHidden = false
+            }
+            sliderALabel.text = "Vignette Center X:"
+            sliderBLabel.text = "Vignette Center Y:"
+            sliderCLabel.text = "Intensity:"
+            sliderDLabel.text = "Radius:"
+
+            sliderA.minimumValue = 0
+            sliderA.maximumValue = Float((scaledImage?.size.width)!)
+            sliderA.value = Float((scaledImage?.size.width)!) / 2
+
+            sliderB.minimumValue = 0
+            sliderB.maximumValue = Float((scaledImage?.size.height)!)
+            sliderB.value = Float((scaledImage?.size.height)!) / 2
+
+            sliderC.minimumValue = 0
+            sliderC.maximumValue = 1
+            sliderC.value = 0.3
+
+            sliderD.minimumValue = 0
+            sliderD.maximumValue = 2000
+            sliderD.value = 100
+        case 2:
+            for i in 0..<allLabels.count {
+                allLabels[i]?.isHidden = false
+                allSliders[i]?.isHidden = false
+            }
+            sliderALabel.text = "Bump Center X:"
+            sliderBLabel.text = "Bump Center Y:"
+            sliderCLabel.text = "Radius:"
+            sliderDLabel.text = "Scale:"
+
+            sliderA.minimumValue = 0
+            sliderA.maximumValue = Float((scaledImage?.size.width)!)
+            sliderA.value = Float((scaledImage?.size.width)!) / 2
+
+            sliderB.minimumValue = 0
+            sliderB.maximumValue = Float((scaledImage?.size.height)!)
+            sliderB.value = Float((scaledImage?.size.height)!) / 2
+
+            sliderC.minimumValue = 0
+            sliderC.maximumValue = 2000
+            sliderC.value = 300
+
+            sliderD.minimumValue = 0
+            sliderD.maximumValue = 1
+            sliderD.value = 0.5
+        case 3:
+            allLabels[0]?.isHidden = false
+            allSliders[0]?.isHidden = false
+
+            sliderALabel.text = "Blur Radius:"
+            sliderA.minimumValue = 0
+            sliderA.maximumValue = 100
+            sliderA.value = 10
+        case 4:
+            allLabels[0]?.isHidden = false
+            allSliders[0]?.isHidden = false
+            allLabels[1]?.isHidden = false
+            allSliders[1]?.isHidden = false
+
+            sliderALabel.text = "Blur Radius:"
+            sliderBLabel.text = "Blur Angle:"
+
+            sliderA.minimumValue = 0
+            sliderA.maximumValue = 100
+            sliderA.value = 20
+
+            sliderB.minimumValue = 0
+            sliderB.maximumValue = 10
+            sliderB.value = 0
+
+        default:
+            break
+        }
+    }
+
```

Comment:
I'd be interested to see how you might refactor this eventually, since there's quite a bit of repetition (but again, that's less important than getting it working).